on_device_training/training_api_demo/mnist_training_example.ipynb

```diff
@@ -0,0 +1,473 @@
+    "        train_loss, _ = model(forward_inputs)\n",
+    "        optimizer.step()\n",
+    "        model.reset_grad()\n",
+    "        losses.append(train_loss)\n",
+    "\n",
+    "    print('Epoch: {},Train Loss: {:.4f}'.format(epoch+1, sum(losses)/len(losses)))\n",
+    "\n",
+    "# Test Loop :\n",
+    "def test(epoch):\n",
+    "    model.eval()\n",
+    "    losses = []\n",
+    "    metric = evaluate.load('accuracy')\n",
+    "\n",
+    "    for batch_idx, (data, target) in enumerate(train_loader):\n",
+    "        forward_inputs = [data.reshape(len(data),784).numpy(),target.numpy().astype(np.int32)]\n",
+    "        test_loss, logits = model(forward_inputs)\n",
+    "        metric.add_batch(references=target, predictions=get_pred(logits))\n",
+    "        losses.append(test_loss)\n",
+    "\n",
+    "    metrics = metric.compute()\n",
+    "    \n",
+    "    print('Epoch: {}, Test Loss: {:.4f}, Accuracy : {:.2f}'.format(epoch+1, sum(losses)/len(losses), metrics['accuracy']*100))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 78,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch: 1,Train Loss: 0.2199\n",
+      "Epoch: 1, Test Loss: 0.1269, Accuracy : 95.92\n",
+      "Epoch: 2,Train Loss: 0.0901\n",
+      "Epoch: 2, Test Loss: 0.0859, Accuracy : 97.03\n",
+      "Epoch: 3,Train Loss: 0.0565\n",
+      "Epoch: 3, Test Loss: 0.0596, Accuracy : 97.92\n",
+      "Epoch: 4,Train Loss: 0.0372\n",
+      "Epoch: 4, Test Loss: 0.0373, Accuracy : 98.75\n",
+      "Epoch: 5,Train Loss: 0.0281\n",
+      "Epoch: 5, Test Loss: 0.0539, Accuracy : 98.29\n"
+     ]
+    }
+   ],
+   "source": [
+    "for epoch in range(5):\n",
+    "    train(epoch)\n",
+    "    test(epoch)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Save Inference model and Run inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 79,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model.export_model_for_inferencing(\"data/inference_model.onnx\",[\"output-0\"])\n",
+    "session = InferenceSession('data/inference_model.onnx',providers=['CPUExecutionProvider'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 80,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Predicted Label :  [7]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# getting one example from test list to try inference.\n",
+    "data = next(iter(test_loader))[0][0].reshape(1,784).numpy()\n",
+    "\n",
+    "input_name = session.get_inputs()[0].name\n",
+    "output_name = session.get_outputs()[0].name \n",
+    "output = session.run([output_name], {input_name: data})\n",
+    "\n",
+    "print(\"Predicted Label : \",get_pred(output[0]))\n",
```

Comment:
get_pred could probably compute the softmax and output the probability score for the chosen result.

Reply:
do we need this for this example? since the point of the example is to show how to use the on device training apis?

Reply:
I think it would be nice to have. But it is fine if we don't have it.